syntax = "proto3";

package server;

service UserService {
  rpc Register(UserInfo) returns (UserResponse);
  rpc Login(UserInfo) returns (UserResponse);
  rpc Logout(UserInfo) returns (UserResponse);
}

service MessageService {
  rpc SendMessage(Message) returns (MessageResponse);
  rpc StreamMessages(Empty) returns (stream Message);
  rpc NotifyUserJoin(Empty) returns (stream UserJoin);
  rpc NotifyUserLeave(Empty) returns (stream UserLeave);
}

message UserInfo{
  string username = 1;
}

message UserResponse{
  string status = 1;
}

message Message {
  string username = 1;
  string text = 2;
}

message MessageResponse{
  string status =1;
}

message UserJoin{
  string username = 1;
}

message UserLeave{
  string username =1;
}

message Empty {}